data-prepper-api/src/test/java/org/opensearch/dataprepper/model/event/JacksonEventTest.java

```diff
@@ -456,6 +457,74 @@ public void testBuild_withStringData() {
         assertThat(event.get("foo", String.class), is(equalTo("bar")));
     }
 
+    @ParameterizedTest
+    @CsvSource({
+        "test-${foo}-string, test-bar-string",
```

Comment:
One other test case we should have - a string without any formatting. It should return the input value.

```
"test-string, test-string"
```